### PR TITLE
Feat/password reset

### PR DESCRIPTION
**Pull request by jnizet**
on top of #36

### CODE REVIEW

File: frontend/src/app/user-edit/user-edit.component.html [hunk 1]
```diff
@@ -37,7 +37,7 @@
 <ngb-alert type="success" [dismissible]="false" *ngIf="createdUser" id="created-user">
   L'utilisateur a été créé. Voici son identifiant et son mot de passe, à lui transmettre.
   <br/>
-  Attention: l'identifiant et le mot de passe sont sensible à la casse&nbsp;!
+  Attention&nbsp;: l'identifiant et le mot de passe sont sensible à la casse&nbsp;!
   <br/>
   Une fois identifié(e), il(elle) pourra changer son mot de passe.
```
Comment (acrepet) [inline frontend/src/app/user-edit/user-edit.component.html]: nice ! Good to feminize the text !

File: frontend/src/app/user.service.ts [hunk 2]
```diff
@@ -68,10 +68,14 @@ export class UserService {
   }
 
   get(id: number): Observable<UserModel> {
-    return this.http.get(`api/users/${id}`).map(response => response.json());
+    return this.http.get(`/api/users/${id}`).map(response => response.json());
   }
 
   update(id: number, user: UserCommand): Observable<void> {
-    return this.http.put(`api/users/${id}`, user).map(() => null);
+    return this.http.put(`/api/users/${id}`, user).map(() => null);
```
Comment (acrepet) [inline frontend/src/app/user.service.ts]: is the '/' behind 'api' mandatory?

File: frontend/src/app/users/users.component.html [hunk 3]
```diff
@@ -1,13 +1,14 @@
...
+<div class="row mb-2 align-items-center user-item" *ngFor="let user of users">
+  <div class="col-sm"><a [routerLink]="['/users', user.id, 'edit']">{{ user.login }}</a></div>
+  <div class="col-sm">
     <span *ngIf="user.admin" class="text-muted">administrateur</span>
     <span *ngIf="!user.admin" class="text-muted">simple utilisateur</span>
   </div>
-  <div class="col text-right">
+  <div class="col-sm text-right">
     <a class="btn btn-secondary" [routerLink]="['/users', user.id, 'edit']">Éditer</a>
+    <a class="btn btn-secondary" [routerLink]="['/users', user.id, 'password-resets']">Réinitilialiser le mot de passe</a>
```
Comment (acrepet) [inline frontend/src/app/users/users.component.html]: réiniti**li**aliser -> réinitialiser

File: frontend/src/app/user.service.ts (hunk 2 above)
Comment (jnizet) [inline frontend/src/app/user.service.ts]: No, not really, but we use it everywhere else.

File: frontend/src/app/password-reset/password-reset.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  const activatedRoute = {
+    snapshot: {data: {user}}
+  };
+
+  beforeEach(() => TestBed.configureTestingModule({
+    imports: [AppModule, RouterTestingModule],
+    providers: [{provide: ActivatedRoute, useValue: activatedRoute}]
+  }));
+
+  it('should reset password', async(() => {
```
Comment (cexbrayat) [inline frontend/src/app/password-reset/password-reset.component.spec.ts]: unnecessary async? unnecessary imports too?

File: frontend/src/app/password-reset/password-reset.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+    const userService = TestBed.get(UserService);
+    const updatedUser = {id: 42, login: 'jb', generatedPassword: 'passw0rd'} as UserWithPasswordModel;
+    spyOn(userService, 'resetPassword').and.returnValue(Observable.of(updatedUser))
+    resetButton.click();
+    fixture.detectChanges();
+
+    expect(fixture.componentInstance.updatedUser).toBe(updatedUser);
+    expect(nativeElement.querySelector('#password-reset-form')).toBeNull();
+    let updatedUserAlert = nativeElement.querySelector('#updated-user');
```
Comment (cexbrayat) [inline frontend/src/app/password-reset/password-reset.component.spec.ts]: let -> const

**Pull request merged**